src/test/java/com/google/sps/QueryByUserIdTest.java

```diff
@@ -0,0 +1,68 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import com.google.sps.UserUtils;
+
+/** tests the IsValidGameName function */
+@RunWith(JUnit4.class)
+public final class QueryByUserIdTest {
```

Comment:
All of those tests are added. I unfortunately couldn't find anything in the testing documentation to test whether the Datastore had been queried. For your last question, when you use FetchOptions.Builder.withLimit(1) like that without a sort on the query, it returns a random value from the datastore, so I think it would return a random user with that id if there were multiple. If we wanted we could add a sort that would change this, but I can't think of why returning one user over another would make sense. Like you said, we'll have to make sure it won't happen when the user entity is created (I think that was already part of the plan, but I'll check).

Reply:
this sounds good. As long as we ensure that there is never any user duplication, we can skip checking for it here. As for Datastore being queried... I can't find anything super easily so I think it's fine if we skip it.